src/components/project/handlers/set-department-id.handler.ts

```diff
@@ -105,8 +105,16 @@ export class SetDepartmentId implements IEventHandler<SubscribedEvent> {
           .where({ 'deptIdNode.value': not(isNull()) })
           .return(collect('deptIdNode.value').as('used')),
       )
-      // Distill to available
-      .with('[id in enumerated where not id in used][0] as next')
+      // Get blacklisted IDs
+      .subQuery((sub) =>
+        sub
+          .match(node('blacklist', 'BlacklistDepartmentId'))
+          .return(collect('blacklist.departmentId').as('blacklisted')),
+      )
+      // Distill to available (excluding both used AND blacklisted)
+      .with(
+        '[id in enumerated where not id in used and not id in blacklisted][0] as next',
```

Comment:
I'd prefer to merge our own used and the externally used IDs into one list first, before this logic happens.

```ts
// Get used IDs
.subQuery((sub) => sub
  .subQuery((sub2) => sub2
    .match([
      node('', 'Project'),
      relation('out', '', 'departmentId', ACTIVE),
      node('deptIdNode', 'Property'),
    ])
    .where({ 'deptIdNode.value': not(isNull()) })
    .return('deptIdNode.value as id')
    .union()
    .match(node('external', 'ExternalDepartmentId'))
    .return('external.departmentId as id'))
  .return(collect('id').as('used'))
)